src/app/data-planes/components/DataplanePolicies.vue

```diff
@@ -250,80 +252,103 @@ function getPolicyEntryConnections(policy: MatchedPolicyType, policyType: Policy
  */
-function getPolicyTypeEntriesFromRules(rules: DataplaneRule[]): PolicyTypeEntry[] {
+function getRuleEntries(rules: DataplaneRule[]): RuleEntry[] {
   // Uses a `Map` to store entries by type so they can be retrieved and updated while iterating over the rules.
-  const policyEntriesByType = new Map<string, PolicyTypeEntry>()
+  const policyTypeEntriesByType = new Map<string, RuleEntry>()
 
   for (const rule of rules) {
-    if (!policyEntriesByType.has(rule.policyType)) {
-      policyEntriesByType.set(rule.policyType, {
+    if (!policyTypeEntriesByType.has(rule.policyType)) {
+      policyTypeEntriesByType.set(rule.policyType, {
         type: rule.policyType,
         connections: [],
       })
     }
 
-    const policyEntry = policyEntriesByType.get(rule.policyType) as PolicyTypeEntry
+    const policyTypeEntry = policyTypeEntriesByType.get(rule.policyType) as RuleEntry
     const policyType = store.state.policyTypesByName[rule.policyType]
-    const connections = getPolicyEntryConnectionsFromRules(rule, policyType)
+    const connections = getRuleEntryConnections(rule, policyType)
 
-    policyEntry.connections.push(...connections)
+    policyTypeEntry.connections.push(...connections)
   }
 
-  const policyTypeEntries = Array.from(policyEntriesByType.values())
+  const policyTypeEntries = Array.from(policyTypeEntriesByType.values())
 
-  policyTypeEntries.sort((policyEntryA, policyEntryB) => policyEntryA.type.localeCompare(policyEntryB.type))
+  policyTypeEntries.sort((policyTypeEntryA, policyTypeEntryB) => policyTypeEntryA.type.localeCompare(policyTypeEntryB.type))
 
   return policyTypeEntries
 }
 
-function getPolicyEntryConnectionsFromRules(rule: DataplaneRule, policyType: PolicyType): PolicyTypeEntryConnection[] {
+function getRuleEntryConnections(rule: DataplaneRule, policyType: PolicyType): RuleEntryConnection[] {
   const { type, service, subset, conf } = rule
 
   // Guards against likely API changes. The response currently contains `"subset": {}` instead of omitting the value or setting it to `null`, but that might change in the future.
   const subsetEntries = subset ? Object.entries(subset) : []
   let sourceTags: LabelValue[] | null
   let destinationTags: LabelValue[] | null
 
+  // Determines source tags.
   if (type === 'clientSubset') {
+    // For client subsets, the source is represented by `subset` (i.e. tags)
     // Sets the wildcard service tag for client subsets without subset values.
     const sourceSubset = subsetEntries.length > 0 ? subsetEntries : [['kuma.io/service', '*']]
-    // For client subsets, the source is represented by `subset` (i.e. tags); for destination subsets and single items, the source is the DPP.
     sourceTags = sourceSubset.length > 0 ? sourceSubset.map(([label, value]) => ({ label, value })) : null
   } else {
+    // For destination subsets and single items, the source is the DPP for which we don’t want to show anything.
     sourceTags = null
```

Comment:
If `sourceTags` was:

```javascript
const sourceTags: LabelValue[] = []
```
instead of 
```javascript
let sourceTags: LabelValue[] | null
```

you could remove this fork/line, then in (and anywhere else): 

https://github.com/kumahq/kuma-gui/pull/551/commits/ed32c8c85549326024ccacfa7d2f6622081274f9#diff-49d71fc15211d9665640235e79525ac0136f3ed7584cd8d56c133f14f73f9fb9R28-R55

```javascript
Array.isArray(rowValue.sourceTags) && rowValue.sourceTags.length > 0
```

You could remove all the isArray checks and just stick with the non-zero length checks. The typing allows `null`, but I also noticed you added that type here too (`RuleEntryConnection`), is there anything preventing these from being just `LabelValue[]` instead of `LabelValue[] | null`? Maybe I missed something? Is there a point that we do something different when this is `null` rather than zero length?

Reply:
I want to explicitly distinguish between "sourceTags are applicable" and "sourceTags are not applicable" by using `null` (which indicates the latter). However, it seems to be the case that the lists of tags can’t actually be empty if they’re applicable for a rule (which can be attributed to the notion that we set the wildcard service tag). Therefore, I can indeed change the semantic of those variables to have `[]` indicate "not applicable".

Technically, I think this is worse to read _semantically_ because an empty list by itself doesn’t mean "this isn’t applicable", but it makes for a slightly improved readability overall of code here.